### PR TITLE
Feature/em user

### PR DESCRIPTION
**Pull request by emorendorff**
# Description

Create search tests/class, finish user class and add tests (short of tests for It 4 pantry), refactor generateUser in scripts.js

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring

# How Has This Been Tested?

- [x] open localHost
- [x] test through Mocha/Chai
- [x] dev tools

# Checklist:

- [x] My code follows the Turing's guidelines of this project
- [x] I have performed a self-review of my own code
- [x] No console error messages

### CODE REVIEW

**Comment by Andrey-1992** [comment]
Great Job wit the user class !

File: src/scripts.js [hunk 1]
```diff
@@ -67,28 +68,42 @@ function searchByTags() {
...
-
 function onStartup() {
-  let userId = (Math.floor(Math.random() * 49) + 1)
-  let newUser = users.find(user => {
-    return user.id === Number(userId);
-  }); // rather than userId === newUser.id
-  user = new User(userId, newUser.name, newUser.pantry)
-  pantry = new Pantry(newUser.pantry)
+  generateUser()
+  pantry = new Pantry(user.pantry)
```
Comment (Eva-Navarrete) [inline src/scripts.js:73]: Nice naming convention

File: src/scripts.js [hunk 2]
```diff
@@ -67,28 +68,42 @@ function searchByTags() {
...
 }
 
+function generateUser() {
+  const randomUserNum = Math.floor(Math.random() * users.length);
+  let matchingUser = users.find((item) => {
+    if (item.id === randomUserNum) {
+      return item;
+    }
+  })
+  user = new User(matchingUser);
```
Comment (Eva-Navarrete) [inline src/scripts.js:85]: Your code looks dry. Nice refactoring.

File: src/user.js [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
-  constructor(id, name, pantry) {
-    this.id = id;
-    this.name = name;
-    this.pantry = pantry;
-    this.favoriteRecipes = [];
+import Search from './search';
 
+class User extends Search {
+  constructor(userData) {
+    super();
```
Comment (Eva-Navarrete) [inline src/user.js:5]: Nice use of super..

**Pull request merged**